gcore/gdal_misc.cpp

```diff
@@ -4557,3 +4557,205 @@ double GDALGetNoDataValueCastToDouble(uint64_t nVal)
+                    if (abySegmentNext[0] == 1 && abySegmentNext[1] == 0x11 &&
+                        abySegmentNext[3] == 2 && abySegmentNext[4] == 0x11 &&
+                        abySegmentNext[6] == 3 && abySegmentNext[7] == 0x11)
+                    {
+                        // no subsampling
+                        osRet += ";subsampling=4:4:4";
+                    }
+                    else if (abySegmentNext[0] == 1 &&
+                             abySegmentNext[1] == 0x22 &&
+                             abySegmentNext[3] == 2 &&
+                             abySegmentNext[4] == 0x11 &&
+                             abySegmentNext[6] == 3 &&
+                             abySegmentNext[7] == 0x11)
+                    {
+                        // classic subsampling
+                        osRet += ";subsampling=4:2:0";
+                    }
+                    else if (abySegmentNext[0] == 1 &&
+                             abySegmentNext[1] == 0x21 &&
+                             abySegmentNext[3] == 2 &&
+                             abySegmentNext[4] == 0x11 &&
+                             abySegmentNext[6] == 3 &&
+                             abySegmentNext[7] == 0x11)
+                    {
+                        osRet += ";subsampling=4:2:2";
+                    }
+                }
+            }
+            else if (markerType == 0xEE && nMarkerSize == 14)
+            {
+                if (VSIFReadL(abyAPP14AdobeMarkerData,
+                              sizeof(abyAPP14AdobeMarkerData), 1, fp) == 1 &&
+                    memcmp(abyAPP14AdobeMarkerData, "Adobe", strlen("Adobe")) ==
+                        0)
+                {
+                    bHasAPP14Adobe = true;
+                }
+            }
+            else if (markerType == 0xDA)
+            {
+                // Start of scan
+                break;
+            }
+            VSIFSeekL(fp, nCurPos + nMarkerSize + 2, SEEK_SET);
+        }
+        std::string osColorspace;
+        if (bHasAPP14Adobe)
+        {
+            if (abyAPP14AdobeMarkerData[11] == 0)
+            {
+                if (nNumComponents == 3)
+                    osColorspace = "RGB";
+                else if (nNumComponents == 4)
+                    osColorspace = "CMYK";
+            }
+            else if (abyAPP14AdobeMarkerData[11] == 1)
+            {
+                osColorspace = "YCbCr";
+            }
+            else if (abyAPP14AdobeMarkerData[11] == 2)
+            {
+                osColorspace = "YCCK";
+            }
+        }
+        else
+        {
+            if (nNumComponents == 3)
+                osColorspace = "YCbCr";
+            else if (nNumComponents == 4)
+                osColorspace = "CMYK";
+        }
+        osRet += ";colorspace=";
+        if (!osColorspace.empty())
+            osRet += osColorspace;
+        else
+            osRet += "unknown";
+    }
+    VSIFSeekL(fp, nSavedPos, SEEK_SET);
```

Comment:
Return value is ignored, is that expected?

Reply:
done